browser/sign.js

```diff
@@ -6,18 +6,18 @@ var BN = require('bn.js')
 var parseKeys = require('parse-asn1')
 var curves = require('./curves.json')
 
-function sign (hash, key, hashType, signType) {
+function sign (hash, key, hashType, signType, tag) {
   var priv = parseKeys(key)
   if (priv.curve) {
-    if (signType !== 'ecdsa') throw new Error('wrong private key type')
+    if (signType !== 'ecdsa' && signType !== 'ecdsa/rsa') throw new Error('wrong private key type')
     return ecSign(hash, priv)
   } else if (priv.type === 'dsa') {
     if (signType !== 'dsa') throw new Error('wrong private key type')
     return dsaSign(hash, priv, hashType)
   } else {
-    if (signType !== 'rsa') throw new Error('wrong private key type')
+    if (signType !== 'rsa' && signType !== 'ecdsa/rsa') throw new Error('wrong private key type')
   }
-
+  hash = Buffer.concat([tag, hash])
```

Comment:
ah it's because we only need the tag when it is an rsa key NOT when it's an ecdsa key so since we can't tell based on the signature type asked for we have to do it based on what rutine they are using, so this code path only is executed for rsa signatures never (ec)dsa